### PR TITLE
Get sign in/out working end to end

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Ended up struggling to test sign up stuff without sign out, so I went ahead in that direction. Hopefully didn't steal too much.

Curious for thoughts here. I believe this is "shippable" with the sagas on mobile enabled at this point. That might save us some pain of having everything be off & then on while we're working.

This is all a good learning exercise for me. I'm starting to get the feeling that we're doing an "in then back out" approach to things.
1. Mirror dispatch & fetch logic in core stuff like account/sign in (double dispatch/select)
2. Move all the "inner pages" onto the local redux store instead of common
3. Go back to the core stuff and remove the web dispatch/select stuff
Maybe this was already obvious to folks...

https://user-images.githubusercontent.com/2731362/185336740-0e9de7ab-45b7-446a-80d5-3b27745bad68.mp4

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Local simulator

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-signinout

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-signinout

File: packages/mobile/src/store/store.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ const createRootReducer = () =>
 
 const sagaMiddleware = createSagaMiddleware({ context: storeContext })
 const middlewares = applyMiddleware(sagaMiddleware)
-const composeEnhancers = composeWithDevTools({ trace: true, traceLimit: 25 })
+const composeEnhancers = composeWithDevTools({ trace: true, traceLimit: 250 })
```
Comment (sliptype) [inline packages/mobile/src/store/store.ts:70]: Yes

File: packages/mobile/src/screens/signon/SignOn.tsx [hunk 2]
```diff
@@ -336,14 +336,26 @@ const SignOn = ({ navigation }: SignOnProps) => {
...
+  const passwordFieldValue: { error: string } = useSelector(getPasswordField)
+  const emailFieldValue: {
+    error: '' | 'inUse' | 'characters'
+    status: 'success' | 'failure'
+  } = useSelector(getEmailField)
+  const accountUser = useSelector(getAccountUser)
+
+  const signedIn = signOnStatus === 'success'
+  const isSigninError = passwordFieldValue.error
+  const emailIsAvailable = emailFieldValue.error !== 'inUse'
```
Comment (sliptype) [inline packages/mobile/src/screens/signon/SignOn.tsx:355]: Is there a reason we aren't using selectors for these anymore? If so can we delete the selectors?

File: packages/mobile/src/store/sagas.ts [hunk 3]
```diff
@@ -17,16 +18,19 @@ export default function* rootSaga() {
   const sagas = [
     // config
     ...backendSagas(),
-    // ...accountSagas(),
+    ...accountSagas(),
```
Comment (sliptype) [inline packages/mobile/src/store/sagas.ts:21]: Did you notice any worse perf with everything enabled like this?

**Review by sliptype — APPROVED** [review]
Looks good! your thinking makes sense, we are gonna have to have the "mirroring" until we drop the webview entirely and then can get rid of it

Also I'm on board with keeping the sagas enabled, but let's keep an eye on perf

File: packages/web/src/common/store/sign-out/sagas.ts [hunk 4]
```diff
@@ -15,6 +16,7 @@ function* watchSignOut() {
   const audiusBackendInstance = yield* getContext('audiusBackendInstance')
   const localStorage = yield* getContext('localStorage')
   yield takeLatest(signOutAction.type, function* () {
+    yield put(resetAccount())
```
Comment (dylanjeffers) [inline packages/web/src/common/store/sign-out/sagas.ts:19]: was this just a bug you were seeing overall?

**Review by dylanjeffers — APPROVED** [review]
awesome changes man, this is happening

File: packages/mobile/src/screens/signon/SignOn.tsx (hunk 2 above)
Comment (raymondjacobson) [reply packages/mobile/src/screens/signon/SignOn.tsx:355]: these are derived values from the result of the selector

File: packages/mobile/src/store/sagas.ts (hunk 3 above)
Comment (raymondjacobson) [reply packages/mobile/src/store/sagas.ts:21]: Across the app no. There's a slight hiccup when pressing the sign in button, but i think fix-forward makes sense for it. Can disable if it's a bad call once it's on the staging app

File: packages/mobile/src/components/sign-out-confirmation-drawer/SignOutConfirmationDrawer.tsx [hunk 5]
```diff
@@ -37,16 +38,18 @@ const useStyles = makeStyles(({ spacing }) => ({
...
   const styles = useStyles()
   const dispatchWeb = useDispatchWeb()
+  const dispatch = useDispatch()
   const { clearHistory } = useSearchHistory()
 
   const { onClose } = useDrawerState(MODAL_NAME)
 
   const handleSignOut = useCallback(() => {
-    dispatchWeb(signOut)
+    dispatch(signOut({}))
```
[the remark this answers is not in the document]
Comment (raymondjacobson) [reply packages/mobile/src/components/sign-out-confirmation-drawer/SignOutConfirmationDrawer.tsx:47]: ts was complaining b/c it's an action creator vs. some js garbo actions. both technically work, but this is more "correct" and I didn't want to change the other

File: packages/web/src/common/store/sign-out/sagas.ts (hunk 4 above)
Comment (raymondjacobson) [reply packages/web/src/common/store/sign-out/sagas.ts:19]: yeah, present on web too, but less impactful

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-signinout

**Pull request merged**